.github/workflows/go-check.yaml

```diff
@@ -0,0 +1,47 @@
+name: Go related checks
+
+on:
+  pull_request: {}
+  push:
+    branches:
+      - master
+
+jobs:
+  go-mod:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.14.2
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Check module vendoring
+        run: |
+          go mod tidy
+          go mod vendor
+          git diff --exit-code
+
+  generate-k8s-api:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.14.2
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          path: src/github.com/${{ github.repository }}
+      - name: Install protobuf dependencies
+        run: |
+          sudo apt install -y protobuf-compiler libprotobuf-dev
+      - name: Set GOPATH
+        run: |
+          echo "##[set-env name=GOPATH;]$GITHUB_WORKSPACE"
+        shell: bash
+      - name: Check k8s generated files
+        run: |
+          cd src/github.com/${{ github.repository }}
+          make generate-k8s-api
```

Comment:
This step is not enough, one needs to delete all `zz_generated.deepcopy.go` files

Reply:
I assume deletion will help to detect dangling files i.e. the file should be not generated but already checked in ?

Reply:
@sayboras I just remember, there's a script doing this :facepalm: https://github.com/cilium/cilium/blob/master/contrib/scripts/check-k8s-code-gen.sh feel free to use the script or to delete if you embed the code into the github action.

>I assume deletion will help to detect dangling files i.e. the file should be not generated but already checked in ?

yes that's correct

Reply:
awesome, I will re-use the script, as other might run this script locally before pushing